File: src/values/u16.rs
pub struct ValueU16 {
    value: Option<u16>,
}

impl ValueU16 {
    pub fn init_empty() -> ValueU16 {
        return ValueU16 { value: None };
    }
    
    pub fn init_with(value: u16) -> ValueU16 {
        return ValueU16 { value: Some(value) };
    }
    
    pub fn show(&self, end: char) {
        if let Some(val) = self.value {
            print!("{val}{end}");
        } else {
            print!("null{end}");
        }
    }
    
    pub fn is_null(&self) -> bool {
        return self.value.is_none();
    }
    
    pub fn get_type_representation(&self) -> &str {
        return "u16";
    }
    
    pub fn get_value(&self) -> Option<u16> {
        return self.value;
    }
}

File: src/values/u16.cpp
#include "values/u16.hpp"
using namespace common::values;

ValueU16::ValueU16()
    : value(std::nullopt) {
    
}

ValueU16::ValueU16(uint16_t value)
    : value(value) {
    
}

void ValueU16::show(char end) const {
    if (this->value)
        std::cout << "u16<" << this->value.value() << '>' << end;
    else
        std::cout << "u16<null>" << end;
}

bool ValueU16::isNull() const {
    return !this->value.has_value();
}

std::string ValueU16::getTypeRepr() const {
    return "u16";
}

ValueKind ValueU16::getKind() const {
    return ValueKind::U16;
}

std::optional<uint16_t> ValueU16::getValue() const {
    return this->value;
}